src/app/header/header.component.ts

```diff
@@ -13,16 +16,34 @@ export class HeaderComponent implements OnInit, OnDestroy {
   userName: string = '';
   authUserSubscribtion?: Subscription;
 
-  constructor(private authService: AuthenticationService, private router: Router) {}
+  constructor(
+    private authService: AuthenticationService, 
+    private router: Router, 
+    private store: Store<fromApp.AppState>
+    ) {}
 
   ngOnInit(): void {
-    this.authUserSubscribtion = this.authService.getUserSubject()
+
+    this.store.select('auth')
+      .pipe(
+        map(authState => authState.name)
+      )
       .subscribe(user => {
-        this.userName = user.name.first
-    });
+        this.userName = user;
+      })
+
+    if(window.localStorage.getItem('name') && window.localStorage.getItem('token')) {
```

Comment:
Can be used without _window_ - just _localStorage_

Reply:
ok, done